tests/src/test/scala/pureconfig/EnumerationsSuite.scala

```diff
@@ -72,6 +72,9 @@ class EnumerationsSuite extends BaseSuite {
   }
 
   it should "not allow deriving readers, writers and full converters for enumerations encoded as sealed traits whose subclasses are not all case objects" in {
+    sealed trait Entity
+    case class Person(name: String, surname: String) extends Entity
+    case class Place(name: String, lat: Double, lon: Double) extends Entity
```

Comment:
This was declared here because these types weren't accessible from this suite, to the derivations below were not compiling for the wrong reasons.

Reply:
Good catch! Can we add a message regex to `illTyped` to prevent it from happening again?

Reply:
Added a message regex in fe4e08b. We should add an `@implicitNotFound` for the enumeration helper type classes, but I'll leave that for another PR.